### PR TITLE
fix(xamlreader): res-dict parsing

### LINKED ISSUE #13099: XamlReader fails to parse add resources to custom ResourceDictionaries with a nested ResourceDictionary property
### Current behavior

Given the following custom `ResourceDictionary` class:

```cs
namespace MyTestApp;
public class TestCustomResourceClass : ResourceDictionary
{
	public ResourceDictionary MyDict { get; set; }
}
```

The following XAML will fail to parse when attempting to add a resource to the nested `MyDict` `ResourceDictionary`

```xml
<tests:TestCustomResourceClass xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
							   xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
							   xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
							   xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
							   xmlns:tests="using:MyTestApp"
							   mc:Ignorable="d">
	<tests:TestCustomResourceClass.MyDict>
		<ResourceDictionary>
			<Color x:Key="MyColor">Red</Color>
		</ResourceDictionary>
	</tests:TestCustomResourceClass.MyDict>
</tests:TestCustomResourceClass>

```

Results in a NullReferenceException at [this line](https://github.com/unoplatform/uno/blob/02e639a9f6bfaf6f7d1f75df0677cf985dd9a159/src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs#L698)

### Expected behavior

Should work

### Works on UWP/WinUI

Yes

### LINKED ISSUE #13100: XamlReader fails to parse ResourceDictionary.MergedDictionaries
### Current behavior
Given the following XAML
```xml
<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
					xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
	<ResourceDictionary.MergedDictionaries>
		<ResourceDictionary>
			<Color x:Key="MyColor">Red</Color>
		</ResourceDictionary>
	</ResourceDictionary.MergedDictionaries>
</ResourceDictionary>
```

XamlReader fails with

```
System.InvalidOperationException: The property System.Collections.Generic.IList`1[Windows.UI.Xaml.ResourceDictionary] MergedDictionaries type does not provide an Add method (Line 8:41
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.ProcessMemberElements(Object instance, XamlMemberDefinition member, PropertyInfo propertyInfo, Object rootInstance)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.ProcessNamedMember(XamlObjectDefinition control, Object instance, XamlMemberDefinition member, Object rootInstance)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.LoadObject(XamlObjectDefinition control, Object rootInstance, Object component, Boolean createInstanceFromXClass)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.ProcessMemberElements(Object instance, XamlMemberDefinition member, PropertyInfo propertyInfo, Object rootInstance)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.ProcessNamedMember(XamlObjectDefinition control, Object instance, XamlMemberDefinition member, Object rootInstance)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.LoadObject(XamlObjectDefinition control, Object rootInstance, Object component, Boolean createInstanceFromXClass)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.AddCollectionItems(Object collectionInstance, IEnumerable`1 nonBindingObjects, Object rootInstance)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.ProcessMemberElements(Object instance, XamlMemberDefinition member, PropertyInfo propertyInfo, Object rootInstance)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.ProcessNamedMember(XamlObjectDefinition control, Object instance, XamlMemberDefinition member, Object rootInstance)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.LoadObject(XamlObjectDefinition control, Object rootInstance, Object component, Boolean createInstanceFromXClass)
   at Windows.UI.Xaml.Markup.Reader.XamlObjectBuilder.Build(Object component, Boolean createInstanceFromXClass)
   at Windows.UI.Xaml.Markup.XamlReader.Load(String xaml)
   at Uno.UI.Demo.Samples.Playground.Update()
```

### Expected behavior
Should work

### Works on UWP/WinUI
Yes

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): closes #13099, closes #13100, re: #13121

## PR Type
What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?
- fixed two instances of crashes when parsing res-dict
- fixed one instance of crash when parsing setter with attached property

### <samp>🤖 Generated by Copilot at 82e43c6</samp>

This pull request enhances the xaml parsing and loading functionality in Uno.UI, by refactoring the test code, fixing some bugs, and adding new features. It introduces the `XamlHelper` class as a shared utility for xaml loading, and improves the handling of `ResourceDictionary` elements and dependency properties in the `XamlObjectBuilder` class. It also simplifies and reorders some using directives in the test files.

## PR Checklist
Please check if your PR fulfills the following requirements:
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
n/a

### CODE REVIEW

File: src/Uno.UI.RuntimeTests/Helpers/XamlHelper.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Windows.UI.Xaml.Markup;
+
+namespace Uno.UI.RuntimeTests.Helpers;
+internal class XamlHelper
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Helpers/XamlHelper.cs:7]: nit:

```suggestion

internal static class XamlHelper
```

Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Helpers/XamlHelper.cs:7]: This is almost a copy from the one in unit tests right?

**Review by Youssef1313 — APPROVED** [review]

File: src/Uno.UI.RuntimeTests/Helpers/XamlHelper.cs (hunk 1 above)
Comment (Xiaoy312) [reply src/Uno.UI.RuntimeTests/Helpers/XamlHelper.cs:7]: yea, it would be best if we unify it in uno.ui

Comment (Xiaoy312) [reply src/Uno.UI.RuntimeTests/Helpers/XamlHelper.cs:7]: not almost... it is the same class
the legacy one in uitest didnt got the patch with xmlns'-ed property
this one has it patched

**Review by ajpinedam — APPROVED** [review]
🔥

File: src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs [hunk 2]
```diff
@@ -618,97 +629,70 @@ private void ProcessMemberElements(object instance, XamlMemberDefinition member,
...
-							&& resourceTargetType == null
-						)
+						if (propertyInfo.GetMethod == null)
 						{
-							throw new InvalidOperationException($"No target type was specified (Line {member.LineNumber}:{member.LinePosition}");
+							throw new InvalidOperationException($"The property {propertyInfo} does not provide a getter (Line {member.LineNumber}:{member.LinePosition}");
 						}
 
-						if ((resourceKey ?? resourceTargetType) is { } key)
+						var rd = (ResourceDictionary)propertyInfo.GetMethod.Invoke(instance, null)!;
```
Comment (kazo0) [inline src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs:669]: I may be missing something somewhere else but wouldn't this give me `null` if I had something like:

```cs
public class MyResDict : ResourceDictionary
{
	public ResourceDictionary MyInnerDict { get; set; }
}
```

with this xaml

```xml
<MyResDict>
	<MyResDict.MyInnerDict>
		<Color x:Key="SomeColorResource">Red</Color>
	</MyResDict.MyInnerDict>
</MyResDict>
```

unless I made it `public ResourceDictionary MyInnerDict { get; set; } = new()`?

Comment (kazo0) [reply src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs:669]: Oh nevermind I see :)

**Review by kazo0 — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs (hunk 2 above)
Comment (Xiaoy312) [reply src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs:669]: this is also a failing condition in winui. so i guess it is fine to leave it like this.

**Comment by MartinZikmund** [comment]
@Xiaoy312 it seems some unit tests are still failing on this one

**Review by jeromelaban — APPROVED** [review]

**Comment by MartinZikmund** [comment]
@mergifyio backport legacy/4x

**Pull request merged**